recipes/boost/all/conanfile.py

```diff
@@ -1317,7 +1317,7 @@ def add_defines(library):
                 cxx_flags.append("-fembed-bitcode")
         if self._with_stacktrace_backtrace:
             flags.append(f"-sLIBBACKTRACE_PATH={self.dependencies['libbacktrace'].package_folder}")
-        if self._stacktrace_from_exception_available and is_apple_os(self) and str(self.settings.compiler.libcxx) == "libc++":
+        if self._stacktrace_from_exception_available and is_apple_os(self) and (str(self.settings.compiler.libcxx) == "libc++" or "x86" not in str(self.settings.arch)):
```

Comment:
Should this error be suppressed for `libc++` on non-intel platforms? From the error message it suggests it should only be suppressed for `libgcc`

Reply:
> Should this error be suppressed for libc++ on non-intel platforms? 

Yes, on Mac M1 does not work (armv8 -> armv8). Please, check the links in the PR description.

> You will also get this error when cross compiling for an ARM based linux platform. Suggest removing the is_apple_os

Good catch, I'll update it.

Reply:
@planetmarshall updated to only suppress in case building non-x86. Tested locally with Mac M1 -> x86_64 + libc++ and is working without the define.